prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -508,17 +588,27 @@ def canonicalize_dataset_automatically(
             )
             logger.info("Unnecessary columns removed")
 
-            dataset_transformer = PromptBasedDatasetTransformer()
-            canonicalized_dataset = dataset_transformer.transform_data(
-                prompt_spec=prompt_spec,
-                dataset=full_dataset["train"],
+            dataset_transformer = PromptBasedDatasetTransformer(
                 num_points_to_transform=num_points_to_transform,
+                max_allowed_failed_transforms=self.max_allowed_failed_transforms,
             )
+            inputs, outputs = dataset_transformer.transform_data(
+                prompt_spec, dataset=full_dataset["train"]
+            )
+            canonicalized_dataset = self.make_dataset_from_samples(inputs, outputs)
             logger.info("Data transformation completed")
 
-            example_rows = json.dumps(canonicalized_dataset["train"][0], indent=4)
+            if (
+                canonicalized_dataset
+                and "train" in canonicalized_dataset
+                and len(canonicalized_dataset["train"]) > 0
+            ):
+                example_rows = json.dumps(canonicalized_dataset["train"][0], indent=4)
 
-            logger.info(f"Transformed dataset. Example row:\n{example_rows}\n")
+                logger.info(f"Transformed dataset. Example row:\n{example_rows}\n")
+            else:
+                dataset_name = top_dataset_info["dataset_name"]
+                logger.info(f"{dataset_name} exceed max allowed transforms..")
```

Comment:
I am guessing you mean max allowed failed transforms here. In that case, shouldn't we be returning None?

Reply:
fixed!